### PR TITLE
Add Account Types

### LINKED ISSUE #18: Add account types
Create account types.

Three tiers of users
Admin
Authors
Users

Admin - full control
Author - can create articles and comments
User - can write comments

### PR DESCRIPTION
**Pull request by Jordan141**
Resolves #18 

This PR will add accounts types and the associated functionality.

Types are:
- Author
- User

This PR will not change how the admin system works.

### CODE REVIEW

**Comment by michaelKurowski** [comment]
- core author / moderator
- author
- user

Normal author needs to wait for core author/moderator acceptance when posting an article

**Comment by michaelKurowski** [comment]
Maybe we can just use some external system for comments?

**Comment by Jordan141** [comment]
Ready for review

File: app/app.js [hunk 1]
```diff
@@ -76,7 +76,7 @@ app.use(require('express-session')({
     secret: 'denmarkisbetterthanswedenandfinland',
     resave: false,
     saveUninitialized: false,
-    cookie: { secure: true, httpOnly: true, sameSite: true}
+    cookie: { secure: false, httpOnly: true, sameSite: true} //Enable secure to true while on HTTPS
```
Comment (michaelKurowski) [inline app/app.js:79]: That should probably be somehow exposed to config or as PROD environmental variable setting?

File: app/routes/articles.js [hunk 2]
```diff
@@ -30,10 +30,53 @@ router.post('/', isLoggedIn, (req, res) => {
...
 //NEW - Show form to create new article
-router.get('/new', isLoggedIn, (req, res) => {
+router.get('/new', isLoggedIn, hasAuthorRole, (req, res) => {
     res.render('articles/new.ejs')
 })
 
+//APPROVE List Article Route
+router.get('/approve', isLoggedIn, (req, res) => {
+    if(!req.user.isAdmin) {
+        req.flash('Oops! Something went wrong!')
```
Comment (michaelKurowski) [inline app/routes/articles.js:40]: Not authorized instead?

File: app/routes/articles.js [hunk 3]
```diff
@@ -30,10 +30,53 @@ router.post('/', isLoggedIn, (req, res) => {
...
+    Article.find({isApproved: false}, (err, articles) => {
+        if(err) return res.sendStatus(500)
+        return res.render('articles/approve', {articles, currentUser: req.user})        
+    })
+})
+
+//APPROVE Show Article Route
+router.get('/approve/:id', isLoggedIn, (req, res) => {
+    if(!req.user.isAdmin || !req.params.id) {
+        req.flash('error', 'Oops! Something went wrong!')
```
Comment (michaelKurowski) [inline app/routes/articles.js:53]: Not authorized instead?

File: app/app.js (hunk 1 above)
Comment (michaelKurowski) [reply app/app.js:79]: It's easy to forget about it

Comment (Jordan141) [reply app/app.js:79]: There's a lot of things that needs changed when we've implemented SSL, adding these all to the env would make quite a messy file. Or I could add the env variable of SSL (boolean) and determine the values / behaviour based off that, however that would need to be it's own PR. What do you think?

File: app/routes/articles.js (hunk 2 above)
Comment (Jordan141) [reply app/routes/articles.js:40]: I don't want the standard user to be aware of the hidden URL, hence the normalized error message.

File: app/routes/articles.js (hunk 3 above)
Comment (Jordan141) [reply app/routes/articles.js:53]: See above

File: app/app.js (hunk 1 above)
Comment (michaelKurowski) [reply app/app.js:79]: No, I mean one PROD flag, that turns the all the security things ON (that are turned off as we develop it)

Comment (michaelKurowski) [reply app/app.js:79]: SSL flag is also an option.  We shouldn't have to track all these comments

Comment (Jordan141) [reply app/app.js]: I've changed it to secure: DEV_MODE ? false : true, instead of secure: !DEV_MODE for readability purposes.

**Pull request merged**